Marks sunpy.io readers private (#6895)
sunpy/sunpy: changelog/6895.deprecation.rst

> - ``sunpy.io.cdf`` sub-module has been marked depreciated, and will be removed in version 5.1.
> This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
> 
> - ``sunpy.io.file_tools`` sub-module has been marked depreciated , and will be removed in version 5.1.
> This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
> 
> - ``sunpy.io.jp2`` sub-module has been marked depreciated , and will be removed in version 5.1.
> This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.

Reviewer: ```suggestion
``sunpy.io.cdf``, ``sunpy.io.file_tools`` and ``sunpy.io.jp2`` sub-modules have been depreciated, and will be removed in version 5.1.

This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
```
